File: content/04/code/java/15--step15-move-givens-into-background/shouty/src/main/java/shouty/Person.java
package shouty;

import java.util.ArrayList;
import java.util.List;

public class Person {
    private final List<String> messagesHeard = new ArrayList<String>();
    private final Network network;

    public Person(Network network) {
        this.network = network;
        network.subscribe(this);
    }

    public List<String> getMessagesHeard() {
        return messagesHeard;
    }

    public void shout(String message) {
        network.broadcast(message);
    }

    public void hear(String message) {
        messagesHeard.add(message);
    }
}

File: content/05/code/dotnet/05--step05-location-specified-in-data-table/Shouty/Shouty/Person.cs
using System;
using System.Collections.Generic;

namespace Shouty
{
    public class Person
    {
        private readonly Network network;
        private readonly List<string> messagesHeard = new List<string>();
        public int Location { get; }

        public Person(Network network, int location)
        {
            this.network = network;
            this.Location = location;
            network.Subscribe(this);
        }

        public void Shout(string message)
        {
            network.Broadcast(message, Location);
        }

        public IList<string> GetMessagesHeard()
        {
            return messagesHeard;
        }

        public void Hear(string message)
        {
            messagesHeard.Add(message);
        }
    }
}
